pg_search/src/postgres/visibility_checker.rs

```diff
@@ -15,18 +15,30 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+use std::sync::Arc;
+
 use crate::postgres::utils;
 use pgrx::pg_sys;
 
+use parking_lot::Mutex;
+
 /// Helper to manage the information necessary to validate that a "ctid" is currently visible to
 /// a snapshot
-pub struct VisibilityChecker {
+#[derive(Clone)]
+pub struct VisibilityChecker(Arc<Mutex<VisibilityCheckerInner>>);
+
+struct VisibilityCheckerInner {
     scan: *mut pg_sys::IndexFetchTableData,
     snapshot: pg_sys::Snapshot,
     tid: pg_sys::ItemPointerData,
 }
 
-impl Drop for VisibilityChecker {
+// SAFETY: `VisibilityChecker` is not actually `Send`... because ~nothing in Postgres' API is
+// Send. But this bound is required due to Tantivy's API, which wants to be able to send
+// `(Segment)Collector`s to background threads.
```

Comment:
... and the reason this is okay for us is because we do not use tantivy's threaded executor.